src/provider/model/Model.ts

```diff
@@ -136,37 +136,51 @@ export class Model {
     }
 
     /**
-     * Gets apps that can handle an intent
+     * Get apps that can handle an intent and optionally with specified context type
      *
-     * Includes windows that are not in the app directory but have registered a listener for it
+     * Includes windows that are not in the app directory but have registered a listener for it if contextType is not specified
      * @param intentType intent type
+     * @param contextType context type
      */
-    public async getApplicationsForIntent(intentType: string): Promise<Application[]> {
+    public async getApplicationsForIntent(intentType: string, contextType?: string): Promise<Application[]> {
```

Comment:
I'm wondering if in the context-provided case, we should include
- Running non-directory apps that have registered a listener for the intent
- Running directory apps that have registered a listener for the intent, where the app directory says they don't handle that intent at all

Those are both cases where we know an app handles an intent (because it has that listener) but don't know which contexts it handles for that intent (because the app directory doesn't tell us). Should we, for consistency, handle those the same as we'd handle an app whose context list for that intent is undefined?

This would still respect the constraint that specifying a `contextType` should narrow down the result. And avoids the weird situation that FDC3 could say that an app supports a particular intent, but would say it supports no particular context types for that intent.

Thoughts?

Reply:
I recall discussing these specific cases with @pjbroadbent . He thinks we should not include them when the context is provided but the intent-context information about the app is unknown (in non-app directory case) or unreported (in directory case). My personal opinion is the same as Phil's but happy to change it to that if there's consensus for the opposite.

Reply:
👍